File: tests/e2e/fixtures/unsigned_integers/mysql/java/src/main/java/com/example/db/com/example/db/queries/Querier.java
package com.example.db.queries;

import com.example.db.models.UnsignedValues;
import java.sql.Connection;
import java.sql.SQLException;
import java.util.List;
import javax.sql.DataSource;

public final class Querier {
    private final DataSource dataSource;

    public Querier(DataSource dataSource) {
        this.dataSource = dataSource;
    }

    public void insertUnsignedRow(short u8Val, int u16Val, long u24Val, long u32Val, java.math.BigInteger u64Val) throws SQLException {
        try (Connection conn = dataSource.getConnection()) {
            Queries.insertUnsignedRow(conn, u8Val, u16Val, u24Val, u32Val, u64Val);
        }
    }

    public List<UnsignedValues> getUnsignedRows() throws SQLException {
        try (Connection conn = dataSource.getConnection()) {
            return Queries.getUnsignedRows(conn);
        }
    }
}

File: tests/e2e/fixtures/unsigned_integers/mysql/kotlin/src/main/kotlin/com/example/db/com/example/db/queries/Querier.kt
package com.example.db.queries

import com.example.db.models.UnsignedValues
import javax.sql.DataSource

class Querier(private val dataSource: DataSource) {

    fun insertUnsignedRow(u8Val: Short, u16Val: Int, u24Val: Long, u32Val: Long, u64Val: java.math.BigInteger): Unit =
        dataSource.connection.use { conn -> Queries.insertUnsignedRow(conn, u8Val, u16Val, u24Val, u32Val, u64Val) }

    fun getUnsignedRows(): List<UnsignedValues> =
        dataSource.connection.use { conn -> Queries.getUnsignedRows(conn) }
}
